itests/org.openhab.core.tests/src/main/java/org/eclipse/smarthome/core/cache/ExpiringCacheAsyncTest.java

```diff
@@ -49,19 +49,21 @@ public void testFetchValue() throws InterruptedException, ExecutionException {
 
         // We expect an immediate result with the value 10.0
         assertEquals(10.0, t.getValue(s).get(), 0.0);
+        verify(s, times(1)).get();
         // The value should be valid
         assertFalse(t.isExpired());
 
         // We expect an immediate result with the value 10.0, but not additional call to the supplier
         assertEquals(10.0, t.getValue(s).get(), 0.0);
         verify(s, times(1)).get();
 
-        // Wait
+        // Wait enough to be sure that the value is expired
         try {
-            Thread.sleep(100);
+            Thread.sleep(125);
```

Comment:
When the test extends `org.eclipse.smarthome.test.java.JavaTest` it can use `waitForAssert` which makes it less prone to timing.

Reply:
Good to know. I now use it to replace the Thread.sleep.